lib/cn/cn.c

```diff
@@ -933,58 +935,65 @@ cn_open(
     }
 
     ctx.cn = cn;
-    ctx.nodemap = map_create(cn->cp->fanout);
+    ctx.nodemap = map_create(CN_FANOUT_MAX);
     if (ev(!ctx.nodemap)) {
         err = merr(ENOMEM);
         goto err_exit;
     }
 
     map_insert_ptr(ctx.nodemap, 0, cn->cn_tree->ct_root);
```

Comment:
I was confused why we're creating the root node manually.  I guess it's to cover the case where CNDB has no kvsets in the root node.  Might add a short comment for that.

Reply:
Yep, ahead of you!  :-)

Reply:
cn_tree_create() creates the root node.  We can upend this in a different ticket if you want...

Reply:
Added a comment, done.

Reply:
I think it's fine to create manually up front.  The callback will create a root if needed, but if the root node is empty, the it wouldn't.  In which case we'd have to check and then create.  So I like it as is.  But I think a comment `/* create root node in case cndb has no root node kvsets */`  would be helpful.  We can also wrap all use of `ctx` behind `ctx_init` and `ctx_free` functions that is paired with the callback.